hardhat.config.ts

```diff
@@ -15,11 +15,27 @@ import dotenv from "dotenv";
 
 dotenv.config();
 
-// const WALLET_PRIVATE_KEY = process.env["WALLET_PRIVATE_KEY"];
-const INFURA_API_KEY = process.env["INFURA_API_KEY"];
+enum PROVIDER {
+  INFURA = "infura",
+  ALCHEMY = "alchemy",
+}
+
+// const WALLET_PRIVATE_KEY = process.env["WALLET_PRIVATE_KEY"] ?? undefined;
+const INFURA_API_KEY = process.env["INFURA_API_KEY"] ?? "";
+const ALCHEMY_API_KEY = process.env["ALCHEMY_API_KEY"] ?? "";
+const PREFERRED_PROVIDER =
+  process.env["INFURA_API_KEY"]?.toLowerCase() === PROVIDER.INFURA
+    ? PROVIDER.INFURA
+    : PROVIDER.ALCHEMY;
```

Comment:
```suggestion
  process.env["INFURA_API_KEY"]?.toLowerCase() === PROVIDER.INFURA
    ? (PROVIDER.INFURA ?? "")
    : (PROVIDER.ALCHEMY ?? "");
```

Not sure if this is valid but I think we should add an empty string in case they are not available.